identity_credential/src/revocation/status_list_2021/status_list.rs

```diff
@@ -0,0 +1,160 @@
+  }
+
+  /// Returns the number of entries.
+  #[allow(clippy::len_without_is_empty)]
+  pub const fn len(&self) -> usize {
+    self.0.len() * 8
+  }
+
+  /// Returns the status of the entry at `index` without bound checking.
+  /// ## Panic:
+  /// * if `index` is greater than or equal to `self.len()`.
+  const fn get_unchecked(&self, index: usize) -> bool {
+    let (i, offset) = Self::entry_index_to_store_index(index);
+    self.0[i] & (0b1000_0000 >> offset) != 0
+  }
+
+  /// Sets the status of the `index`-th entry to `value`.
+  /// ## Panic:
+  /// * if `index` is greater than or equal to `self.len()`.
+  fn set_unchecked(&mut self, index: usize, value: bool) {
+    let (i, offset) = Self::entry_index_to_store_index(index);
+    if value {
+      self.0[i] |= 0b1000_0000 >> offset
+    } else {
+      self.0[i] &= 0b0111_1111 >> offset
+    }
+  }
+
+  /// Returns the status of the `index`-th entry, if it exists.
+  pub fn get(&self, index: usize) -> Result<bool, StatusListError> {
+    (index < self.len())
+      .then_some(self.get_unchecked(index))
+      .ok_or(StatusListError::IndexOutOfBounds)
+  }
+
+  /// Sets the status fo the `index`-th entry to `value`.
+  pub fn set(&mut self, index: usize, value: bool) -> Result<(), StatusListError> {
+    if index < self.len() {
+      self.set_unchecked(index, value);
+      Ok(())
+    } else {
+      Err(StatusListError::IndexOutOfBounds)
+    }
+  }
+
+  /// Attempts to parse a [`StatusList2021`] from a string, following the
+  /// [StatusList2021 expansion algorithm](https://www.w3.org/TR/2023/WD-vc-status-list-20230427/#bitstring-expansion-algorithm).
+  pub fn try_from_encoded_str(s: &str) -> Result<Self, StatusListError> {
+    use flate2::read::GzDecoder;
+    use identity_core::convert::Base;
+    use identity_core::convert::BaseEncoding;
+
+    let compressed_status_list =
+      BaseEncoding::decode(s, Base::Base64Url).or(Err(StatusListError::InvalidEncoding(s.to_owned())))?;
+    let status_list = {
+      use std::io::Read;
+
+      let mut decompressor = GzDecoder::new(&compressed_status_list[..]);
+      let mut status_list = vec![];
+      decompressor
+        .read_to_end(&mut status_list)
+        .or(Err(StatusListError::InvalidEncoding(s.to_owned())))?;
+
+      StatusList2021(status_list.into_boxed_slice())
+    };
+
+    Ok(status_list)
+  }
+
+  /// Encode this [`StatusList2021`] into its string representation following
+  /// [StatusList2021 generation algorithm](https://www.w3.org/TR/2023/WD-vc-status-list-20230427/#bitstring-generation-algorithm).
+  pub fn into_encoded_str(self) -> String {
+    use flate2::write::GzEncoder;
+    use flate2::Compression;
+    use identity_core::convert::Base;
+    use identity_core::convert::BaseEncoding;
+
+    let compressed_status_list = {
+      use std::io::Write;
+
+      let mut compressor = GzEncoder::new(vec![], Compression::best());
+      compressor.write_all(&self.0).expect("Out of memory");
```

Comment:
Actually it won't even return an error, it will straight up panic if out of memory (although it's not even mentioned in the docs). Check out the [vec's write implementation](https://doc.rust-lang.org/src/std/io/impls.rs.html#395-398)